src/components/tooltip/Marker.js

```diff
@@ -714,6 +717,43 @@ class MarkerTooltipContents extends React.PureComponent<Props> {
       implementationFilter,
       zeroAt
     );
+
+    // FileIO markers can belong to different threads.
+    let threadDetail = null;
+    const data = marker.data;
+    if (data && data.type === 'FileIO' && data.threadId !== undefined) {
+      const theadId = data.threadId;
+      const occurringThreadName = threadIdToNameMap.get(theadId);
+      threadDetail = [
+        <TooltipDetail label="Recording Thread" key="recording">
+          {threadName}
+        </TooltipDetail>,
+      ];
+
+      // If we have the thread information of the occurring thread, then show.
+      // Otherwise, only show the thread ID.
+      if (occurringThreadName !== undefined) {
+        threadDetail.push(
+          <TooltipDetail
+            label="Occurring Thread"
+            key="occurring"
+          >{`${occurringThreadName} (${theadId})`}</TooltipDetail>
+        );
+      } else {
+        threadDetail.push(
+          <TooltipDetail label="Occurring Thread ID" key="occurring">
+            {theadId}
+          </TooltipDetail>
+        );
+      }
+    } else {
+      threadDetail = [
+        <TooltipDetail label="Thread" key="thread">
+          {threadName}
+        </TooltipDetail>,
+      ];
+    }
```

Comment:
What about moving this code to a method `this._getThreadDetail` (like we do for `_getUrl` for example)? That would make the `render` method clearer.

Reply:
yeah right, another method would make it look better. Added it!